tests/unit/test_coreneuron.py

```diff
@@ -0,0 +1,191 @@
+"""Test coreneuron simultion solver and direct mode
+In file emode, spikes dat file is created by CoreNEURON, check the file content
+In direct mode, spikes vectors are returned to NEURON, no dat file created, check the spikes vector
+"""
+
+from pathlib import Path
+
+import numpy as np
+import numpy.testing as npt
+import pytest
+
+from tests.utils import check_directory
+
+from neurodamus import Neurodamus
+from neurodamus.core.configuration import SimConfig
+
+ref_gids = np.array([1, 2, 3, 1, 2, 3, 1, 2, 3])  # 1-based
+ref_timestamps = np.array([5.1, 5.1, 5.1, 25.1, 25.1, 25.1, 45.1, 45.1, 45.1])
+
+
+@pytest.mark.parametrize(
+    "create_tmp_simulation_config_file",
+    [
+        {
+            "simconfig_fixture": "ringtest_baseconfig",
+            "extra_config": {
+                "target_simulator": "CORENEURON",
+                "inputs": {
+                    "pulse": {
+                        "module": "pulse",
+                        "input_type": "current_clamp",
+                        "delay": 5,
+                        "duration": 50,
+                        "node_set": "RingA",
+                        "represents_physical_electrode": True,
+                        "amp_start": 10,
+                        "width": 1,
+                        "frequency": 50,
+                    }
+                }
+            }
+        }
+    ],
+    indirect=True,
+)
+def test_coreneuron_filemode(create_tmp_simulation_config_file):
+    """Test the default coreneuron run, filemode
+    1. coreneuron_input data folder, sim.conf, report.conf created and deleted after simulation
+    2. populations_offset.dat created for reporting
+    3. check the "out.dat" generated by coreneuron
+    """
+    n = Neurodamus(create_tmp_simulation_config_file)
+    coreneuron_data = Path(SimConfig.coreneuron_datadir)
+    check_directory(coreneuron_data)
+    assert (Path(SimConfig.output_root) / "sim.conf").exists()
+    assert (Path(SimConfig.output_root) / "report.conf").exists()
+    assert (Path(SimConfig.output_root) / "populations_offset.dat").exists()
+
+    n.run()
```

Comment:
just to be sure:

should generated files be the same also for:
```
n.run(False)
```

Reply:
yes, the same. It is the clean up step that executes the deletion.